lightly/active_learning/agents/agent.py

```diff
@@ -120,36 +188,40 @@ def query(self, sampler_config: SamplerConfig, al_scorer: Scorer = None) -> Tupl
+
+        # handle illogical stopping condition
         if sampler_config.n_samples < len(self.labeled_set):
-            warnings.warn("ActiveLearningAgent.query: The number of samples which should be sampled "
-                          "including the current labeled set "
-                          "(sampler_config.n_samples) "
-                          "is smaller than the number of samples in the current labeled set."
-                          "Skipping the sampling and returning the previous labeled set.")
+            warnings.warn(
+                f'ActiveLearningAgent.query: The number of samples ({sampler_config.n_samples}) is '
+                f'smaller than the number of preselected samples ({len(self.labeled_set)}).'
+                'Skipping the active learning query and returning the previous labeled set.'
+            )
             return self.labeled_set, []
 
-        # calculate scores
+        # calculate active learning scores
+        scores_dict = None
         if al_scorer is not None:
-            no_unlabeled_samples = len(self.unlabeled_set)
-            no_samples_with_predictions = len(al_scorer.model_output)
-            if no_unlabeled_samples != no_samples_with_predictions:
-                raise ValueError(f"The scorer must have exactly as many samples as in the unlabeled set,"
-                                 f"but there are {no_samples_with_predictions} predictions in the scorer,"
-                                 f"but {no_unlabeled_samples} in the unlabeled set.")
+            no_query_samples = len(self.query_set)
+            no_query_samples_with_scores = len(al_scorer.model_output)
+            if no_query_samples != no_query_samples_with_scores:
+                raise ValueError(
+                    f'Number of query samples ({no_query_samples}) must match '
+                    f'the number of predictions ({no_query_samples_with_scores})!'
+                )
             scores_dict = al_scorer.calculate_scores()
-        else:
-            scores_dict = None
 
         # perform the sampling
         new_tag_data = self.api_workflow_client.sampling(
             sampler_config=sampler_config,
             al_scores=scores_dict,
-            preselected_tag_id=self.preselected_tag_id,
-            query_tag_id=self.query_tag_id)
-
-        # set the newly chosen tag as the new preselected_tag_id and update the sets
-        self.preselected_tag_id = new_tag_data.id
-        self._set_labeled_and_unlabeled_set(new_tag_data)
+            preselected_tag_id=self._preselected_tag_id,
+            query_tag_id=self._query_tag_id
+        )
+
+        # update the old preselected_tag
+        self._old_preselected_tag_bitmask = self._preselected_tag_bitmask
+        # set the newly chosen tag as the new preselected_tag
+        self._preselected_tag_id = new_tag_data.id
+        self._preselected_tag_bitmask = self._get_preselected_tag_bitmask()
 
         return self.labeled_set, self.added_set
```

Comment:
Perhaps we should just return nothing and let the user access the sets by him/herself. Then no documentation for the return values is need.
It is also a bit unlogical, that the query_set, which is needed for predicting the new scores, is not returned.